nuxeo-core/nuxeo-core/src/main/java/org/nuxeo/ecm/core/blob/ColdStorageHelper.java

```diff
@@ -80,6 +84,11 @@ public static DocumentModel moveContentToColdStorage(CoreSession session, Docume
         DocumentModel documentModel = session.getDocument(documentRef);
         log.debug("Move to cold storage the main content of document: {}", documentModel);
 
+        if (!session.hasPermission(documentRef, SecurityConstants.WRITE_COLD_STORAGE)) {
+            throw new NuxeoException(String.format("The document: %s cannot be moved to cold storage", documentRef),
```

Comment:
You could be more precise on the message about the missing required permission.

Reply:
For debugging purposes I understand that it would be interesting to have more details.
But, if I am not mistaken, this exception will be sent to client side and, for security purposes, we should not provide too much details (IMHO).

Reply:
Yes we try to avoid too many details in the exceptions especially when related to security. It can be worth adding `log.debug` statements with the details if you feel then can be useful.

However in this particular case "cannot be moved to cold storage" seems sufficient to understand the issue.

Reply:
Added!
Thanks for your input @efge and @ataillefer!